ew_agents/osint_tools.py

```
@@ -175,6 +196,126 @@ def classify_image_content_theme(image_url: str, associated_text: Optional[str]
         "associated_text": associated_text
     }
 
+# =============================================================================
+# Fallback Analysis Functions
+# =============================================================================
+
+def analyze_narrative_patterns_fallback(text: str) -> dict:
+    """Fallback narrative pattern analysis using keyword matching"""
+    import re
```

Ruff (0.12.2)

`re` imported but unused

Remove unused import: `re`

(F401)